main.py

```diff
@@ -77,33 +97,57 @@ def export_streams_to_clipboard(self):
 
     def add_new_stream(self, *args, stream_url=None, stream_quality=cfg[CONFIG_QUALITY]):
         """Adds a new player for the specified stream in the grid."""
+
         if not stream_url:
             stream_url, ok = QtWidgets.QInputDialog.getText(self, "Stream input", "Enter the stream URL:")
 
             if not ok:
                 return
 
+        # Give some feedback to the user
+        self.grid.addWidget(self.loading, self.coordinates.x, self.coordinates.y)
+        self.loading.show()
+
+        # Run the rest on a seperate thread to be able to show the loading feedback
+        # Also helps a lot with lag
+        threading.Thread(target=self._add_new_stream, args=(stream_url, stream_quality)).start()
+
+    def _add_new_stream(self, stream_url=None, stream_quality=cfg[CONFIG_QUALITY]):
+        """Fetches qualities and if possible adds a frame to the main window."""
         try:
             stream_options = self.streamlink_session.streams(stream_url)
 
             if stream_quality not in stream_options:
-                stream_quality, ok = self._get_user_quality_preference(stream_options)
-
-                if not ok:
-                    return
+                self.fail_add_stream.emit(AddStreamError.default_quality_missing, (stream_options, stream_url, stream_quality))
```

Comment:
`AddStreamError.default_quality_missing` should become `AddStreamError.DEFAULT_QUALITY_MISSING`